source/val/validate_memory.cpp

```diff
@@ -1430,6 +1430,85 @@ spv_result_t ValidateAccessChain(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
+spv_result_t ValidateRawAccessChain(ValidationState_t& _,
+                                    const Instruction* inst) {
+  std::string instr_name = "Op" + std::string(spvOpcodeString(inst->opcode()));
+
+  if (!_.features().raw_access_chains) {
+    return _.diag(SPV_ERROR_INVALID_DATA, inst)
+           << "Generating raw access chains requires capability "
+              "RawAccessChainsEXT";
+  }
+
+  // The result type must be OpTypePointer.
+  const auto result_type = _.FindDef(inst->type_id());
+  if (spv::Op::OpTypePointer != result_type->opcode()) {
+    return _.diag(SPV_ERROR_INVALID_ID, inst)
+           << "The Result Type of " << instr_name << " <id> "
+           << _.getIdName(inst->id()) << " must be OpTypePointer. Found Op"
+           << spvOpcodeString(result_type->opcode())
+           << '.';
+  }
+
+  // The pointed storage class must be valid.
+  const auto storage_class = result_type->GetOperandAs<spv::StorageClass>(1);
+  if (storage_class != spv::StorageClass::Workgroup &&
```

Comment:
Workgroup is not listed as a valid storage class in the [spec](https://htmlpreview.github.io/?https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/NV/SPV_NV_raw_access_chains.html). Is that a spec bug or tooling bug?

Reply:
It was a bug here. Removed it.